server/src/Infrastructure/DependencyInjection.cs

```diff
@@ -21,7 +21,9 @@ public static IServiceCollection AddInfrastructure(this IServiceCollection servi
             throw new InvalidOperationException("The JWT signing key must be configured.");
         }
 
-        services.AddDbContext<ApplicationDbContext>(options => options.UseSqlServer(configuration.GetConnectionString("DefaultConnection")));
+        services.AddDbContext<ApplicationDbContext>(options =>
+        options.UseSqlServer(configuration.GetConnectionString("DefaultConnection"))
+        .AddInterceptors(new AuditableEntityInterceptor()));
```

Comment:
https://learn.microsoft.com/en-us/ef/core/logging-events-diagnostics/interceptors

it suggests overriding OnConfiguring method (in ApplicationDbContext) and registering interceptors there.
also it is stateless, so can be made static (2nd example in the link)

Reply:
Should we create also two contexts or to add this logic to ApplicationDbContext ? For example:
public class AuditableEntityInterceptorContext: ApplicationDbContext 
{
     // here same logic as 2 example from The link 
}

@arminasbrazenas

Reply:
Override the method of existing appdbcontext